### PR TITLE
Add Azure DNS zone and records

### PR DESCRIPTION
**Pull request by RMcVelia**
## Description

Add Azure DNS zone and associated records

Base resources created using
`make dns deploy-domain-resources AUTO_APPROVE=1`

To check zone deployment
`make dns domains-infra-plan`

_environment_domains/config/trp_qa* is just an example, it will be updated once we know what permanent domains will be ceated_

Once the zone is created, can then request NS record from parent domain

## Trello Card Link

https://trello.com/c/hnhS0Hbn/404-irp-create-dns-zone-in-azure-dns-via-terraform

### CODE REVIEW

File: terraform/custom_domains/environment_domains/main.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Used to create domains to be managed by front door.
+module "domains" {
+  for_each              = var.hosted_zone
+  source                = "git::https://github.com/DFE-Digital/terraform-modules.git//domains/environment_domains?ref=0.5.4"
```
Comment (saliceti) [inline terraform/custom_domains/environment_domains/main.tf:4]: Can we use ref=stable

File: terraform/custom_domains/environment_domains/main.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  environment           = each.value.environment_short
+  host_name             = each.value.origin_hostname
+  multiple_hosted_zones = var.multiple_hosted_zones
+  null_host_header      = try(each.value.null_host_header, false)
+  cached_paths          = try(each.value.cached_paths, [])
+}
+
+# Takes values from hosted_zone.domain_name.cnames (or txt_records, a-records). Use for domains which are not associated with front door.
+module "dns_records" {
+  source      = "git::https://github.com/DFE-Digital/terraform-modules.git//dns/records?ref=0.5.2"
```
Comment (saliceti) [inline terraform/custom_domains/environment_domains/main.tf:18]: Can we use ref=stable

File: terraform/custom_domains/environment_domains/variables.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
+variable hosted_zone {
+  type = map(any)
+  default = {}
+}
+
+variable "multiple_hosted_zones" {
```
Comment (saliceti) [inline terraform/custom_domains/environment_domains/variables.tf:6]: Can this be removed as there is only 1 zone

File: terraform/custom_domains/environment_domains/config/trp_qa.tfvars.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+  "multiple_hosted_zones": true,
```
[the remark this answers is not in the document]
Comment (RMcVelia) [reply terraform/custom_domains/environment_domains/config/trp_qa.tfvars.json]: done

File: terraform/custom_domains/environment_domains/main.tf (hunk 1 above)
Comment (RMcVelia) [reply terraform/custom_domains/environment_domains/main.tf]: done

File: terraform/custom_domains/environment_domains/main.tf (hunk 2 above)
Comment (RMcVelia) [reply terraform/custom_domains/environment_domains/main.tf]: done

File: terraform/custom_domains/environment_domains/variables.tf (hunk 3 above)
Comment (RMcVelia) [reply terraform/custom_domains/environment_domains/variables.tf]: done

**Review by saliceti — APPROVED** [review]

**Pull request merged**